oio/container/lifecycle.py

```diff
@@ -182,28 +184,27 @@ def apply(self, obj_meta, **kwargs):
             else:
                 yield obj_meta, rule.id, "n/a", "Kept"
 
-    def execute(self, use_precessed_versions=True, **kwargs):
-        """
-        Match then apply the set of rules of the lifecycle configuration
-        on all objects of the container.
-
-        :returns: tuples of (object metadata, rule name, action, status)
-        :rtype: generator of 4-tuples
-        :notice: you must consume the results or the rules won't be applied.
-        """
-        if use_precessed_versions:
-            self.processed_versions = ProcessedVersions()
+    def process_container(self, container, **kwargs):
         for obj_meta in depaginate(
                 self.api.object_list,
                 listing_key=lambda x: x['objects'],
                 marker_key=lambda x: x.get('next_marker'),
                 truncated_key=lambda x: x['truncated'],
                 account=self.account,
-                container=self.container,
+                container=container,
                 properties=True,
                 versions=True,
                 **kwargs):
             try:
+                # save name inside container for API use
+                obj_meta['org_name'] = obj_meta['name']
+                # and reconstruct full path of object if needed
+                if '%2F' in container:
```

Comment:
The delimiter should be a variable (or at least a constant). For specific use cases we may want to change it.